app/models/user_stats.rb

```diff
@@ -10,7 +10,39 @@ def self.exp_for(user)
     self.stats_for(user).exp
   end
 
+  def activity(date_range = nil)
+    {
+        exercises: {
+            solved_count: organization_exercises
+                              .joins(:assignments)
+                              .where(assignments: { top_submission_status: [:passed, :skipped], submitter: user }.merge_if_present(:submitted_at, date_range))
+                              .count,
+            count: organization_exercises.count},
+
+        messages: {
+            count: messages_in_discussions(date_range).count,
+            approved: messages_in_discussions(date_range).where(approved: true).count}
+    }
+  end
+
   def add_exp!(points)
     self.exp += points
   end
+
+  private
+
+  def messages_in_discussions(date_range = nil)
+    Message.joins(:discussion)
+        .where({sender: user.uid, discussions: { organization: organization }}.merge_if_present(:date, date_range))
```

Comment:
I would propose to take advantage of compact and just do:

data_param = {date: date_range}.compact
.....
merge(date_param)

:smile: